dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -172,6 +172,8 @@ internal class RumViewScope(
         if (stopped) return
 
         val context = getRumContext()
+        if (!context.sessionKept) return
```

Comment:
Having this check on each method can be error prone, can't we find a different way, maybe a method that checks if the session is valid an executes an action ? `checkSessionAndExecute`. The ideal way would have been to validate this at the writer level, this way we would never forget to check the session validity if let's say we add a new method in one of the scopes to send an event.